src/select/select_value.rs

```diff
@@ -0,0 +1,35 @@
+use crate::select::JsonPathError;
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum SelectValueType {
+    Undef,
+    Null,
+    Bool,
+    Long,
+    Double,
+    String,
+    Array,
+    Dict,
+}
+
+pub trait SelectValue:
+    std::fmt::Debug + std::cmp::Eq + std::cmp::PartialEq + std::default::Default + std::clone::Clone
```

Comment:
The idiomatic way is to `use std::...` and then use the short names of the traits.